plugins/catalog-backend/src/processing/ProcessorOutputCollector.ts

```diff
@@ -105,8 +105,9 @@ export class ProcessorOutputCollector {
 
       this.deferredEntities.push({ entity, locationKey: location });
     } else if (i.type === 'location') {
+      const presence = i.optional ? 'optional' : 'required';
       const entity = locationSpecToLocationEntity(
-        i.location,
+        { presence, ...i.location },
```

Comment:
Went for backwards compatibility for now, we'd ofc get rid of the `optional` parameter at some point as we look at the processing API again